src/lib/operators/insert.cpp

```diff
@@ -8,94 +8,80 @@
 #include "concurrency/transaction_context.hpp"
 #include "resolve_type.hpp"
 #include "storage/base_encoded_segment.hpp"
+#include "storage/segment_iterate.hpp"
 #include "storage/storage_manager.hpp"
 #include "storage/value_segment.hpp"
 #include "type_cast.hpp"
 #include "utils/assert.hpp"
 
-namespace opossum {
+namespace {
 
-// We need these classes to perform the dynamic cast into a templated ValueSegment
-class AbstractTypedSegmentProcessor : public Noncopyable {
- public:
-  AbstractTypedSegmentProcessor() = default;
-  AbstractTypedSegmentProcessor(const AbstractTypedSegmentProcessor&) = delete;
-  AbstractTypedSegmentProcessor& operator=(const AbstractTypedSegmentProcessor&) = delete;
-  AbstractTypedSegmentProcessor(AbstractTypedSegmentProcessor&&) = default;
-  AbstractTypedSegmentProcessor& operator=(AbstractTypedSegmentProcessor&&) = default;
-  virtual ~AbstractTypedSegmentProcessor() = default;
-  virtual void resize_vector(std::shared_ptr<BaseSegment> segment, size_t new_size) = 0;
-  virtual void copy_data(std::shared_ptr<const BaseSegment> source, ChunkOffset source_start_index,
-                         std::shared_ptr<BaseSegment> target, ChunkOffset target_start_index, ChunkOffset length) = 0;
-};
+using namespace opossum;  // NOLINT
 
 template <typename T>
-class TypedSegmentProcessor : public AbstractTypedSegmentProcessor {
- public:
-  void resize_vector(std::shared_ptr<BaseSegment> segment, size_t new_size) override {
-    auto value_segment = std::dynamic_pointer_cast<ValueSegment<T>>(segment);
-    DebugAssert(value_segment, "Cannot insert into non-ValueColumns");
-    auto& values = value_segment->values();
+void resize_value_segment(std::shared_ptr<BaseSegment> segment, size_t new_size) {
+  const auto value_segment = std::dynamic_pointer_cast<ValueSegment<T>>(segment);
+  Assert(value_segment, "Cannot insert into non-ValueColumns");
```

Comment:
Haha, I just thought about refactoring an hour ago and how the column-to-segment thing was largely without complications...

Reply:
;)